doc/developers/contributing.rst

```diff
@@ -762,6 +762,85 @@ To test code coverage, you need to install the `coverage
 
 3. Loop.
 
+Monitoring performance [*]_
+===========================
+
+When proposing changes to the existing code base, it's important to make sure
+that they don't introduce performance regressions. Scikit-learn uses
+`asv benchmarks <https://github.com/airspeed-velocity/asv>`_ to monitor the
+performance of a selection of common estimators and functions. The benchmark
+suite can be found in the `scikit-learn/asv_benchmarks` directory.
+
+To use all features of asv, you will need either `conda` or `virtualenv`. For
+more details please check the `asv installation webpage
+<https://asv.readthedocs.io/en/latest/installing.html>`_.
+
+First of all you need to install the development version of asv::
+
+  pip install git+https://github.com/airspeed-velocity/asv
+
+and change your directory to `asv_benchmarks/`::
+
+  cd asv_benchmarks/
+
+The benchmark suite is configured to run against your local clone of
+scikit-learn. Make sure it is up to date::
+
+  git fetch upstream
+
+In the benchmark suite, the benchmarks are organized following the same
+structure as scikit-learn. For example, you can compare the performance of a
+specific estimator between master and the branch you are working on::
+
+  asv continuous -b LogisticRegression origin/master HEAD
+
+You can also specify a whole module to benchmark::
+
+  asv continuous -b linear_model origin/master HEAD
+
+You can replace `HEAD` by any branch. By default it will only report the
+benchmarks that have change by at least 10%. You can control this ratio with
+the `-f` flag.
+
+To run the full benchmark suite, simply remove the `-b` flag ::
+
+  asv continuous origin/master HEAD
+
+However this can take up to two hours. The `-b` flag also accepts a regular
+expression for a more complex subset of benchmarks to run.
+
+To run the benchmarks without comparing to another branch, use the `run`
+command::
+
+  asv run -b linear_model HEAD^!
+
+You can also run the benchmark suite using the version of scikit-learn already
+installed in your current Python environment::
+
+  asv run --python=same
+
+It's particulary useful when you installed scikit-learn in editable mode. By
```

Comment:
You can run the benchmarks once in an env where you have it installed in editable mode and then in an env where you have installed from let's say conda to compare the perf between using mkl or openblas.

Also, even though you have it installed in editable mode, by default asv will create an isolated env to run the benchmarks.

I think both mode are useful

Reply:
But is this really related to the editable mode?

What I understand from your comment is that this is useful when one wants to have full control on the dependencies, instead of relying on whatever `asv` will be installing by default?